Build from snapcraft.yaml and use ucdev PPA

alfonsosanchezbeato opened the pull request:
Remove makefile and do special steps from an override-build section
instead.

Take now signed grub and shim form the ucdev/uc-staging-ppa PPA, using
package-repositories. Now we use epochs in the version of
grub-efi-amd64-signed and shim-signed packages, so the build should
always take the ones from the PPA instead of those in the archive, as
`2:<version>` will always be greater than the package version in the
latter. Nonetheless, do an additional static check for this.

Forward-ported from https://github.com/snapcore/pc-amd64-gadget/pull/65

alfonsosanchezbeato committed:
Build from snapcraft.yaml and use ucdev PPAs

Remove makefile and do special steps from an override-build section
instead.

Take now signed grub and shim form the ucdev/uc-staging-ppa PPA, using
package-repositories. Now we use epochs in the version of
grub-efi-amd64-signed and shim-signed packages, so the build should
always take the ones from the PPA instead of those in the archive, as
2:<version> will always be greater than the package version in the
latter.

alfonsosanchezbeato committed:
Check that shim/grub are signed with the UC cert

Forward-ported from 1fa803e

alfonsosanchezbeato committed:
Use CRAFT_* instead of SNAPCRAFT_*

As required by snapcraft 7+.

alfonsosanchezbeato commented:
Note that this will not build until we have jammy binaries in https://launchpad.net/~ucdev/+archive/ubuntu/uc-staging-ppa.

vorlonofportland commented:
> Note that this will not build until we have jammy binaries in https://launchpad.net/~ucdev/+archive/ubuntu/uc-staging-ppa.

Those binaries are there as of yesterday..

alfonsosanchezbeato commented:
> > Note that this will not build until we have jammy binaries in https://launchpad.net/~ucdev/+archive/ubuntu/uc-staging-ppa.
> 
> Those binaries are there as of yesterday..

For focal... not for jammy. For jammy we have a good shim in the old PPA, I would like to upload a new one when we have shim 15.6 signed by Microsoft.

alfonsosanchezbeato commented:
Now that we have the jammy packages, I've rebased this PR.

alfonsosanchezbeato committed:
snapcraft.yaml: pin ucdev PPA

This is needed at the moment as signed grub from PPA needs the
grub-efi-amd64-bin package from the PPA, which can diverge slightly
from the one in the updates pocket.

alfonsosanchezbeato commented:
Somewhat gadgets built using shim/grub from latest in ucdev are failing for FDE, blocking the merge for the moment.

xnox reviewed (changes requested):
Needs fixing in light of https://github.com/snapcore/pc-gadget/pull/81 getting merged

alfonsosanchezbeato committed:
Use makefile for mbr and fixes after rebase

alfonsosanchezbeato commented:
> Needs fixing in light of #81 getting merged

Rebased now

alfonsosanchezbeato committed:
Add assumes with snapd supporting latest shim

alfonsosanchezbeato commented:
Updated by adding "assumes" cc @xnox
